Source/Details/ASPINRemoteImageDownloader.m

```diff
@@ -116,56 +115,67 @@ + (ASPINRemoteImageDownloader *)sharedDownloader NS_RETURNS_RETAINED
-  static dispatch_once_t onceToken;
-  dispatch_once(&onceToken, ^{
-
-    if (preconfiguredPINRemoteImageManager) {
-      sharedPINRemoteImageManager = preconfiguredPINRemoteImageManager;
-    } else {
+  PINRemoteImageManager *manager = nil;
+  // Check that Carthage users have linked both PINRemoteImage & PINCache by testing for one file each
+  if (!(NSClassFromString(@"PINRemoteImageManager"))) {
+    NSException *e = [NSException
+                      exceptionWithName:@"FrameworkSetupException"
+                      reason:@"Missing the path to the PINRemoteImage framework."
+                      userInfo:nil];
+    @throw e;
+  }
+  if (!(NSClassFromString(@"PINCache"))) {
+    NSException *e = [NSException
+                      exceptionWithName:@"FrameworkSetupException"
+                      reason:@"Missing the path to the PINCache framework."
+                      userInfo:nil];
+    @throw e;
+  }
 #if PIN_ANIMATED_AVAILABLE
-      // Check that Carthage users have linked both PINRemoteImage & PINCache by testing for one file each
-      if (!(NSClassFromString(@"PINRemoteImageManager"))) {
-        NSException *e = [NSException
-                          exceptionWithName:@"FrameworkSetupException"
-                          reason:@"Missing the path to the PINRemoteImage framework."
-                          userInfo:nil];
-        @throw e;
-      }
-      if (!(NSClassFromString(@"PINCache"))) {
-        NSException *e = [NSException
-                          exceptionWithName:@"FrameworkSetupException"
-                          reason:@"Missing the path to the PINCache framework."
-                          userInfo:nil];
-        @throw e;
-      }
-      sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
-                                                                alternativeRepresentationProvider:[self sharedDownloader]];
+    sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
+                                                                   alternativeRepresentationProvider:[self sharedDownloader]
+                                                                   imageCache:imageCache];
 #else
-      sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration];
+    manager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
+                                                                 alternativeRepresentationProvider:nil
+                                                                 imageCache:imageCache];
 #endif
-    }
-  });
-  return sharedPINRemoteImageManager;
+  return manager;
 }
 
 - (PINRemoteImageManager *)sharedPINRemoteImageManager
 {
-  return [ASPINRemoteImageDownloader sharedPINRemoteImageManagerWithConfiguration:nil preconfiguredPINRemoteImageManager:nil];
+  if (!sharedPINRemoteImageManager) {
```

Comment:
Why do we need to have this check?

Reply:
Give me a minute to try to remember lol!

If I can't, I will write it off as a brain fart...

Reply:
OK, I took nine, and I got nothing! I am pretty sure I had worked myself into some overcomplicated concurrency scenario that cannot actually exist and wouldn't have been completely helped anyway